---
title: "Pre processing GSE125416"
output: html_notebook
---

GSE125416 is an experiment comparing normal hESC cells and nicotine exposed hESC cells. High rate of cell cycle is expected. It contains two samples:

- GSM3573649	Control_EB scRNA-seq [D];
- GSM3573650	nicotine-exposed_EB scRNA-seq [N].

Data is accessible at [Gene Expression Omnibus - GSE125416 ](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE125416).

Article is accessible at [Stem Cell Reports - Single-Cell RNA Sequencing of Human Embryonic Stem Cell Differentiation Delineates Adverse Effects of Nicotine on Embryonic Development](https://www.sciencedirect.com/science/article/pii/S2213671119300244)

# Read data
We use Seurat to read and preprocess the data.

```{r}
library(Seurat)
filepath = "/home/shaoheng/Documents/data/hESC-droplet/GSE125416/GSM3573649_D_filtered_gene_bc_matrices_h5.h5"
obj = CreateSeuratObject(Read10X_h5(filepath), project='wt')
obj
```

# Filtering

```{r}
obj[["percent.mt"]] <- PercentageFeatureSet(obj, pattern = "^MT-")
VlnPlot(obj, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
```

## Mitochondria and expressed genes
"cells with fewer than 200 or more than 6,000 expressed genes, and a high percentage of mitochondrial genes (>20%) were removed." This is done by `subset`. Should keep 6766 cells.

```{r}
obj <- subset(obj, subset = nFeature_RNA >= 200 & nFeature_RNA <= 6000 & percent.mt < 20)
obj
```

## Normalization
"We next normalized the gene expression values for each cell by the total expression, multiplied by 10,000, and made a log-transformation." This is done by `NormalizeData`.
```{r}
obj <- NormalizeData(obj)
```

## Variable genes
Variable expressed genes across the single cells under cutoff with an average expression of more than 0.0125 and less than 3, and dispersion of more than 0.5, were detected for down-stream analysis. 

```{r}
obj <- FindVariableFeatures(obj, mean.cutoff = c(0.0125, 3), dispersion.cutoff = c(0.5, Inf))
```

## Cell cycle
```{r}
obj <- CellCycleScoring(obj, s.features = cc.genes$s.genes, g2m.features = cc.genes$g2m.genes, set.ident = TRUE)
RidgePlot(obj, features = c("CCND1", "CCNE1", "CCNA2", "CCNB1"), ncol = 2)
```

```{r}
RidgePlot(obj, features = c("MKI67", "TUBB4B"), ncol = 2)
```

## PCA
```{r}
obj <- ScaleData(obj, features = rownames(obj))
```

```{r}
obj <- RunPCA(obj, features = VariableFeatures(object = obj))
DimPlot(obj)
```

```{r}
DimPlot(obj, dims = c(2, 3), pt.size=0.1)
```

```{r}
saveRDS(obj, "~/Documents/data/hESC-droplet/GSE125416/GSM357649-regular.RDS")
```

```{r}
source("../utils/gsea.R")
mat2cls(as.matrix(obj@reductions$pca@cell.embeddings), "~/Documents/data/hESC-droplet/GSE125416/GSM357649-regular-pca.cls")
```

```{r}
source("../utils/hdfrw.R")
mat2hdf(as.matrix(obj@assays$RNA@scale.data), "~/Documents/data/hESC-droplet/GSE125416/GSM357649-regular.h5")
```
